src/lib/OccAnalysis.hs

```diff
@@ -100,6 +100,14 @@ useManyTimes (FV fvs) = FV $ mapNameMapE update fvs where
   -- Note: Could do something more elaborate here to keep track of
   -- which indices of each object are used much, but for now there's
   -- no need, because any Unbounded count will inhibit inlining.
+
+  -- TODO(Oddity) This will also set apparent indexing depth to zero.  The
+  -- indexing depth that occurrence analysis counts is relevant for assessing
+  -- work preservation, in that an inlining might only be work-preserving if it
+  -- leads to canceling against an indexing expression (or several).  However,
+  -- no inlining into a `while` loop is work-preserving in any case, so zero is
+  -- sound.  It does look odd on test cases, though, if the binding is clearly
+  -- indexed inside the while.
```

Comment:
I feel like this loss of precision might be dangerous? From what I understand, you're arguing here that losing the index-depth doesn't matter, because the binder will shows up as having unbounded use. But, as we discussed yesterday, there might be binders that you might want to inline an arbitrary number of times, for as long as they cancel out against the index: broadcasts! Doing the analysis this way might lead us to inline them eagerly, leaving extra loops in the IR.

Reply:
OK, fixed it.